### PR TITLE
Java, Postgres and Oracle Generators

### PR DESCRIPTION
**Pull request by abrievanaardt**

### CODE REVIEW

File: src/main/java/bbd/jportal2/JPortal2Arguments.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return "JPortal2Arguments{" +
+                "logFileName='" + logFileName + '\'' +
+                ", inputDirs=" + inputDirs +
+                ", inputFiles=" + inputFiles +
+                ", builtinSIProcessors=" + builtinSIProcessors +
+                ", templateSIProcessors=" + templateSIProcessors +
```
Comment (dieterrosch) [inline src/main/java/bbd/jportal2/JPortal2Arguments.java]: Can we do this with reflection?

File: src/main/java/bbd/jportal2/Main.java [hunk 2]
```diff
@@ -1,145 +1,50 @@
...
-            System.exit(1);
-        }
+        try {
+            ProjectCompiler pj = ProjectCompilerBuilder.build(arguments);
 
-        //Set this before the logger starts.
-        if (main.logFileName != null)
-            System.setProperty("log.name", main.logFileName);
+            if (Objects.nonNull(pj)) {
+                int rc = pj.compileAll();
```
Comment (dieterrosch) [inline src/main/java/bbd/jportal2/Main.java]: Don’t think this if should be nested?

File: src/main/java/bbd/jportal2/generators/JavaJCCode.java [hunk 3]
```diff
@@ -0,0 +1,883 @@
...
+
+    private static void generateStructs(Table table, String output) {
+        generateStdProcStruct(table, output);
+        generateOtherProcStructs(table, output);
+    }
+
+    private static void generateStdProcStruct(Table table, String output) {
+        try {
+            logger.info("Code: " + output + table.useName() + "Struct.java");
+            OutputStream outFile = new FileOutputStream(output + table.useName() + "Struct.java");
```
Comment (dieterrosch) [inline src/main/java/bbd/jportal2/generators/JavaJCCode.java]: Don’t open the file yourself. Use the methods in the base class.

File: src/main/java/bbd/jportal2/generators/JavaJCCode.java [hunk 4]
```diff
@@ -0,0 +1,883 @@
...
+            try {
+                outData = new PrintWriter(outFile);
```
Comment (dieterrosch) [inline src/main/java/bbd/jportal2/generators/JavaJCCode.java]: As above

File: src/main/java/bbd/jportal2/generators/JavaJCCode.java [hunk 5]
```diff
@@ -0,0 +1,883 @@
...
+                continue;
+            if (!proc.isStd && !proc.hasNoData())
+                generateOtherProcStruct(table, proc, output);
+        }
+    }
+
+    private static void generateOtherProcStruct(Table table, Proc proc, String output) {
+        try {
+            logger.info("Code: " + output + table.useName() + proc.upperFirst() + "Struct.java");
+            OutputStream outFile = new FileOutputStream(output + table.useName() + proc.upperFirst() + "Struct.java");
```
Comment (dieterrosch) [inline src/main/java/bbd/jportal2/generators/JavaJCCode.java]: As above

File: src/main/java/bbd/jportal2/JPortal2Arguments.java (hunk 1 above)
Comment (abrievanaardt) [reply src/main/java/bbd/jportal2/JPortal2Arguments.java]: We should be using Lombok to generate that. The JAR is about 1.1MB though but will allow us to get rid of all the boilerplate code: getters, setters, builders, constructors, equals and hash methods as well as toString.

File: src/main/java/bbd/jportal2/Main.java (hunk 2 above)
Comment (abrievanaardt) [reply src/main/java/bbd/jportal2/Main.java]: The `if` looks nested but the `if` above it has been moved. There's no curly brace, it's a standalone `if`.

Comment (dieterrosch) [reply src/main/java/bbd/jportal2/Main.java]: Ok I see. Please add braces, fix whitespace or write on one line?

File: pom.xml [hunk 6]
```diff
@@ -4,7 +4,7 @@
 
     <groupId>za.co.bbd</groupId>
     <artifactId>jportal2</artifactId>
-    <version>1.0</version>
+    <version>1.0.0-SNAPSHOT</version>
```
Comment (dieterrosch) [inline pom.xml]: Still says snapshot? Is this fixed in the pipelines?

**Pull request merged**